service/gcs/main.go

```diff
@@ -150,6 +164,18 @@ func main() {
 		logrus.WithError(err).Fatal("opengcs::main - failed add gcs pid to gcs cgroup")
 	}
 
+	gcsEfd, err := cgrouputils.RegisterMemoryThreshold(filepath.Join(cgrouputils.MemRoot, "gcs"), gcsLimit)
+	if err != nil {
+		logrus.WithError(err).Fatal("opengcs::main - failed to register memory threshold for gcs cgroup")
+	}
+
+	containerEfd, err := containersControl.OOMEventFD()
+	if err != nil {
+		logrus.WithError(err).Fatal("opengcs::main - failed to retrieve the container cgroups oom eventfd")
+	}
+
+	go readMemoryEvents(gcsEfd, "gcs", gcsLimit)
```

Comment:
Can we add some tests for these two scenarios? The tests might have to be in the hcsshim repo. This could be a followup change.

Reply:
Yes, I didn't know of how to test this in this repo directly but that can be a good test for the shim